packages-node/nodejs-helpers/src/tasks/bypass-export-map.js

```diff
@@ -0,0 +1,193 @@
+    const require = createRequire(import.meta.url);
+    const initialResolvedPath = require.resolve(initialSource, { paths: [exportsDir] });
+    const adjustedResolvedPath = require.resolve(adjustedSource, { paths: [outputDir] });
+    return initialResolvedPath === adjustedResolvedPath;
+  } catch (error) {
+    throw new Error(ERROR_CAN_NOT_RESOLVE_SOURCE);
+  }
+};
+
+/**
+ * Adjusts the source value of an import/export declaration.
+ *   Example: `import sth from source;`
+ * The new source value is calculated relative to the outputDir.
+ * And checked to be resolvable to the same resource with Node.js require.resolve
+ *
+ * https://github.com/jamiebuilds/babel-handbook/blob/master/translations/en/plugin-handbook.md#visitors
+ * @param {string} exportsDir
+ * @param {string} outputDir
+ * @throws ERROR_ADJUSTED_SOURCE_IS_INVALID
+ */
+const getAdjustImportExportVisitor = (exportsDir, outputDir) => ({
+  // @ts-ignore
+  enter({ node }) {
+    const isImportOrExportNode = isImportDeclaration(node) || isExportDeclaration(node);
+    if (!isImportOrExportNode) {
+      return;
+    }
+    // @ts-ignore
+    const { source } = node;
+    const initialSource = source?.value;
+    if (!initialSource) {
+      return;
+    }
+    const isRelativeSource = initialSource?.startsWith('.');
+    if (!isRelativeSource) {
+      return;
+    }
+    const adjustedSource = adjustSource(initialSource, exportsDir, outputDir);
+    if (!resolvesToSameResource(initialSource, adjustedSource, exportsDir, outputDir)) {
+      throw new Error(ERROR_ADJUSTED_SOURCE_IS_INVALID);
+    }
+    // register the new source value
+    source.value = adjustedSource;
+  },
+});
+
+/**
+ * Adjusts relative paths for a given fileName,
+ * and writes it with the same fileName under the outputDir
+ *
+ * @param {string} fileName
+ * @param {string} exportsDir
+ * @param {string} outputDir
+ * @throws ERROR_CAN_NOT_OVERWRITE_EXISTING_FILE
+ */
+const generateAdjustedExportFile = async (fileName, exportsDir, outputDir) => {
+  const outputPath = path.resolve(outputDir, fileName);
+  if (existsSync(outputPath)) {
+    throw new Error(ERROR_CAN_NOT_OVERWRITE_EXISTING_FILE);
+  }
+  const inputPath = path.resolve(exportsDir, fileName);
+  const initialCode = await readFile(inputPath, 'utf-8');
+  const adjustImportExportVisitor = getAdjustImportExportVisitor(exportsDir, outputDir);
+  const adjustedCode = transformCode(initialCode, adjustImportExportVisitor);
+  return writeFile(outputPath, prettify(adjustedCode));
+};
+
+/**
+ * @param {string} packageDir
+ * @param {string} exportMapKey
+ * @param {string} exportMapValue
+ * @returns {Promise<any>}
+ */
+const bypassExportMapItem = async (packageDir, exportMapKey, exportMapValue) => {
+  const exportMapValuePath = path.join(packageDir, exportMapValue);
+  const exportMapKeyPath = path.join(packageDir, exportMapKey);
+  const searchPattern = isDirectorySync(exportMapValuePath)
+    ? path.join(exportMapValuePath, '**', '*.js')
```

Comment:
At some point, we might also support css import assertions. At that point, we can probably extract the generic export map resolver from providence and use it here as well.